bentoml/yatai/validator/deployment_pb_validator.py

```diff
@@ -167,6 +168,28 @@ def _validate_azure_functions_configuration(
                     'Azure Functions min instances must be smaller than max burst',
                 )
 
+    def _validate_deployment_labels(self, deployment_labels, field, value):
+        """ Test label key value schema
+
+        The rule's arguments are validated against this schema:
+        {'type': 'boolean'}
+        """
+        if deployment_labels:
+            pattern = re.compile("^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$")
+            for key in value.keys():
+                if (
```

Comment:
can we abstract this out to a `validate_label` method and reuse it in the other method on the client-side?